### PR TITLE
Add IPC4 HDA playback support

### PR DESCRIPTION
**Pull request by RanderWang**
Now HDA playback can go without any XRUN in IPC4 path.

### CODE REVIEW

File: src/audio/Kconfig [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 menu "Audio components"
 
+config COMP_COPIER
+        bool "COPIER component"
+        default y
+        help
+          Select for COPIER component
```
Comment (cujomalainey) [inline src/audio/Kconfig:9]: what is this? this gives no detail on what this is.

File: src/include/ipc4/base-config.h [hunk 2]
```diff
@@ -103,4 +103,23 @@ struct BaseModuleCfg {
...
+};
+
+struct ipc4_output_pin_format {
+	uint32_t			pin_index; /*!< \brief Index of the pin.*/
+	uint32_t			obs;	   /*!< \brief Specifies output frame size (in bytes).*/
+	struct ipc4_audio_format 		audio_fmt; /*!< \brief Format of the output data.*/
+};
+
+enum ipc4_stream_type {
+    ipc4_pcm = 0,                   ///< PCM stream
```
Comment (cujomalainey) [inline src/include/ipc4/base-config.h:119]: what is ipc4?

File: src/include/ipc4/error_status.h [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+ * \brief IPC4 global definitions.
+ * NOTE: This ABI uses bit fields and is non portable.
+ */
+
+#ifndef __SOF_IPC4_ERROR_STATUS_H__
+#define __SOF_IPC4_ERROR_STATUS_H__
+
+#include <ipc4/pipeline.h>
+
+enum ipc4_status {
```
Comment (cujomalainey) [inline src/include/ipc4/error_status.h:25]: why are we not using linux error codes?

**Review by cujomalainey — COMMENTED** [review]
Is there a reason this appears to have been rushed through without review? IPC is a core component so this does not appear to be very community friendly.

**Comment by ranj063** [comment]
> Is there a reason this appears to have been rushed through without review? IPC is a core component so this does not appear to be very community friendly.

@cujomalainey this hasnt been merged to main. Its a WIP branch for testing IPC4

**Comment by cujomalainey** [comment]
ah I missed that, thanks, I wonder if we can scope codeowners to a specific branch

**Comment by paulstelian97** [comment]
> ah I missed that, thanks, and apologies. I wonder if we can scope codeowners to a specific branch

The only way is to have a different CODEOWNERS file on said branch, but that's not friendly with merging back to `main`.

**Comment by lgirdwood** [comment]
This is just one of my topic branches that some of us are working on, so not worth codeowners as it wont go into main until cleaned up.

**Pull request merged**